### ISSUE TITLE
Merging Allentown and Chicago

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1475]
In GitLab by @kdjstudios on May 15, 2019, 10:17

**Submitted by:** "Haydee Yepez" <haydee.yepez@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/8364272

**Server:** Internal

**Client/Site:** Multiple

**Account:** All

**Issue:** 

We need to merge Allentown and Chicago billing this coming month, what do we need to do for that?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 15, 2019, 10:26

I wrote:

> Thank you for reaching out. I believe you are requesting to perform the same type of merge that we had completed with Memphis and West Memphis in the middle of last year. if so, there is in summary I believe three major tasks/steps, outlined below, yet it is a complex process. I believe we will want to schedule a call with all parties involved to outline a timeline, milestones, and expected results to ensure this is properly executed. I have included Tim on this email also to encase he has any additional insight. 
> 
> 1. Operations: VCC update all accounts to include site number prefix (OA and Motion); VCC Exception report must be blank; Create activity mapping document for SAB tech team. 
> 2. VCC Tech: Merge both sites onto the same servers; merge historical databases to maintain reporting; ops rechecks with VCC Exception report. 
> 3. SAB Tech: Move all accounts from one site to another using the activity mapping document from ops form step one; Dry run import and check SAB Exception report; operational sign off on approval of completion.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 20, 2019, 12:26

Tim wrote:

> We shouldn't need #2 at all since Allentown and Chicago are already on the same VCC server and they may have even setup the site prefixes when we first moved them to VCC.

Haydee wrote:

> Kyle- what time are we able to have this call?  With Tim stating we do not do #2 what is the next step?

I wrote:

> I have Tuesday the 21st anytime from 11am to 5pm MT and also on Friday the 24th from 11am to 5pm MT available this week. I am unsure what time(s) Tim has available.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 12, 2019, 12:30

@smasih @amishra @pchaudhary @tim.traylor 

There has been multiple meetings and communication regarding merging the two sites into one. I have below the update of steps and progress so far to my knowledge. In summary we are still waiting on operations to complete their steps. However, I wanted to make sure we will have a dev server available to perform the merge and test a billing cycle on. Our deadline to have this released to production will be the June 21st. This way they can run their June 22nd billing cycle.  


**Operations:** (Have yet to hear if this has been completed) VCC update all accounts to include site number prefix (OA and Motion); VCC Exception report must be blank; Create activity mapping document for SAB tech team. 
- (Have yet to hear if this has been completed) Attached is a list of expected accounts that will need to be updated; please note there may be more, but this should be a good start. These we believe are the accounts that are not prefixed with the Site number. We will look into updating the PayGo accounts via the back end as there is quite a few; prefix with "061" to save operations time. (I have just followed up with Keith and Tim about the back-end updates)
- Double check that the VCC codes are the same, if not please list those on the activity mapping document. 
- (I provided an draft mapping document for operations to fill in, have yet to receive this, I attached the file again here)  Mapping document is for any codes (Late fees, credits, debits, others, etc) that will need to be changed from the Chicago site and what the fee needs to be changed to on the Allentown side. 
- (Have yet to hear if this has been completed) Exception report: could skip past any sections that do not deal with the Account Name, Account Number, and Resource ID. 
 * We will want operations to look further into the "Northrup Grumman" account to determine if the employee entries are able to be removed from the back end. (Separate ticket)
- (Based upon your email sounds like this is still in progress) Please be sure to update clients and inform them of the changes to their invoices (usage dates, service dates, etc)

**SAB Tech:** (Waiting for operations completion of steps above and approval) Move all accounts from one site to another using the activity mapping document from ops form step one; Dry run import and check SAB Exception report; operational sign off on approval of completion.
- (Will be completed after the merge of all accounts) Change site name from "Allentown' to "Dyad". Moving Chicago accounts to the Allentown. 
- We will want to 'copy' the updates to the development server and run the import and double check the SAB exception report.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 14, 2019, 13:02

@smasih @amishra @pchaudhary @tim.traylor 

What all is needed for us to get the accounts moved over and setup on the dev server? This has a deadline of June 22nd, so we need to ensure all is ready for operations to test and then we can release on June 22nd to production.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 17, 2019, 06:05

@kdjstudios Kyle, 
1. Will the team need to do the account move on Saturday during the release cycle?
1. When Aavaz moves the accounts, is Chicago immediately going onto the same billing cycle as Allentown?
1. The team will need time to code and test the mapping, so if they don't get that to us immediately it will cause a delay in the merge. (Taxes, late charges, activities, fees)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 17, 2019, 12:03

@tim.traylor @smasih @amishra @pchaudhary 

1. We will want to do a move and testing this week. The production release should be this Saturday the 22nd.
2. Yes.
3. Office, Taxes, and Late fees should be moved over. The billing code mapping document should be provided by Kim soon, She was working on it on Friday.

EDIT: Here is the mapping document provided by Kim. Please let me know if anything else is needed. 

[Copy+of+Allentown+and+Chicago+Merge+Mapping.xlsx](/uploads/9ee3328b83a1b5e6896dea9816d6fc26/Copy+of+Allentown+and+Chicago+Merge+Mapping.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 18, 2019, 08:16

@kdjstudios @tim.traylor 

We have successfully mapped the recurring fees & activities of the **Chicago** accounts & moved them to the **Allentown**.

* We have mapped the same offices for the accounts of the Chicago if the Allentown had that office name otherwise we have added the **A** office for all of the other accounts of Chicago.

* Please check on the Merge Server & let us know if there's any change we have to do on our end.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 18, 2019, 08:16

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2019, 08:59

@pchaudhary 

May you please confirm which server this is on? I just looked up an account under the Chicago site and then tired to look up that same account under the Allentown site, but it was not found?

![Merge](/uploads/cdcdc7918dc7e7bde035a511a909d31b/Merge.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 19, 2019, 02:07

@kdjstudios 

We have fixed that.

As discussed on the huddle with Tim, that we have to move the accounts not copied to the Allentown, so we have now moved all of the accounts of the Chicago to the Allentown Site.

**Offices**

* If Allentown has the same office name which Chicago Account has, then we linked that office to the account otherwise we have linked the office **A** of the Allentown for the remaining accounts.

**Additional Fee**

* If Allentown has the same Additional Fee code which Chicago Account has, then we linked that additional fee to the account otherwise we have created the new additional fee of the same details for the Allentown which Chicago Additional fee has & linked that fee as the recurring to the account.

**Activity**

* If Allentown has the same Activity Fee code which Chicago Account has, then we linked that activity fee to the account otherwise we have created the new activity fee of the same details for the Allentown which Chicago Activity fee has & linked that fee to the account.


**Note-** 
1.  Please ask the Ops to change the Office for the accounts if they need different.
1.  Now, Chicago does not have any accounts.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2019, 08:25

@pchaudhary @smasih @tim.traylor @amishra 

Thanks for the update. I have informed operations to confirm the move. 

I wrote:

> The support team was able to use the mapping document to move the Chicago accounts to the Allentown site. This is setup on our development server listed below for your teams review and approval before releasing to production. Also once your teams have finished with the exception list, we would recommend to run a billing export report and do a test billing cycle on the development server to insure there are no issues.
> 
> Please note: If Allentown had the same office name for which a Chicago Account has, then the support team linked that office to the account otherwise the support team linked the Chicago Account to Allentown's office A.
> 
> Once we have your approval that all the accounts were moved correctly in SAB and that operations was able to confirm a billing cycle with no issues we will have the support team publish these changes to production. 

Overall Update:

**SAB Tech:** (Waiting for operations completion of steps above and approval) Move all accounts from one site to another using the activity mapping document from ops form step one; Dry run import and check SAB Exception report; operational sign off on approval of completion.
> Move is completed, waiting on operational sign off.

- (Will be completed after the merge of all accounts) Change site name from "Allentown' to "Dyad". Moving Chicago accounts to the Allentown.

> Still need to rename site to Dyad; this can be done when we move to Production.

- We will want to 'copy' the updates to the development server and run the import and double check the SAB exception report.

> Waiting for operations to confirm they are able to import and billing on merge server.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 20, 2019, 24:37

thanks for the update

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2019, 10:09

@smasih @amishra @pchaudhary @tim.traylor 

I just had a phone call with operations today. They should be testing the billing cycle and if all goes as planned we will be ready to move these updates to production.

We just wanted to confirm, that these updates will not change any of their existing staged fees that they have been inputting on production?

EDIT:

They found a few accounts that needed to have their resource IDs updated on merge in order to get everything to work correctly. May we please push all of account updates on merge to production. Also please update the site name of allentown to "Dyad" too please. 

Alina wrote:

> We have looked at everything in merging.  It is approved to move to production.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 25, 2019, 07:31

@kdjstudios @tim.traylor 

We have successfully moved the Chicago Accounts to the Allentown. Below are some of the points we need to mention:

* We have moved all of the accounts (Active/Terminated) from Chicago to Allentown, so from now on there will be no accounts can be found under the Chicago site.


**Offices**

* If Allentown has the same office name which Chicago Account has, then we linked that office to the account otherwise we have linked the office **A** of the Allentown for the remaining accounts.

**Additional Fee**

* If Allentown has the same Additional Fee code which Chicago Account has, then we linked that additional fee to the account otherwise we have created the new additional fee with the same details for the Allentown which Chicago Additional fee has & linked that fee as the recurring to the account.

**Activity**

* If Allentown has the same Activity Fee code which Chicago Account has, then we linked that activity fee to the account otherwise we have created the new activity fee with the same details for the Allentown which Chicago Activity fee has & linked that activity to the account.

**Late Charges**

* There are 2 Late Charges named as Late Charge & Late Charge 2 of the site Chicago but the Allentown has none of them. So, we have created 2 new late charges of the same name for the Allentown site & linked those with the respected newly moved accounts.

**Billing Periods** 

* There are 15 Billing Periods under the Chicago Site but we didn't find any of the Chicago account's recurring fee linked to any of the billing periods. So, we didn't create any of the Billing Period under the Allentown Site.

**Taxes**

* There is no Taxes under the Chicago Site.

**Holidays**

* We didn't change anything new with the Allentown Holiday Fee.

**Sub Billing Cycle**

* There are 3 Billing Cycle under the Chicago site.
  1.  Monthly
  1.  NCSM 1st Cycle Accounts - 0 Account
  1.  NCSM 15th Cycle Accounts  - 1 Account (N1013 NCSM- Aviem)

* The last 2 Billing Cycle names are also presented under the Allentown Site. So, we created only 1 new billing cycle named as **Monthly** of which Last Billing Cycle Date
updated as the Last Billing Cycle Date of the **Monthly** Billing Cycle of the Chicago Site which is `01-June-2019`. 
* We have also linked all of the Chicago accounts having the Monthly Billing cycle to the newly created Billing Cycle under the Chicago site.
* There is only 1 account of the Chicago Site having the **NCSM 15th Cycle Accounts** which is now linked to the Allentown's **NCSM 15th Cycle Accounts**.



**Note-** 
1.  Please ask the Ops to change the Office for the accounts if they need different.
1.  Now, Chicago does not have any accounts.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2019, 08:42

@pchaudhary 

I wrote:

> Hello Alina and everyone,
> 
> The support team was able to perform the following updates to production. You should now be able to begin your billing cycle process on Production. Please take extra caution and double check everything is working correctly. 
> 
> 1. We have successfully moved the Chicago Accounts to the Allentown. Below are some of the points we would like to add further clarification on.
> 2. We have moved all of the accounts (Active/Terminated) from Chicago to Allentown, so from now on there will be no accounts can be found under the Chicago site.
> 
> Offices:
> 
> 1. If Allentown has the same office name which Chicago Account has, then we linked that office to the account otherwise we have linked the office A of the Allentown for the remaining accounts.
> 
> 
> Additional Fee:
> 
> 1. If Allentown has the same Additional Fee code which Chicago Account has, then we linked that additional fee to the account otherwise we have created the new additional fee with the same details for the Allentown which Chicago Additional fee has & linked that fee as the recurring to the account. The support team was able to use the mapping document provided to aid with this. 
> 
> Activity:
> 
> 1. If Allentown has the same Activity Fee code which Chicago Account has, then we linked that activity fee to the account otherwise we have created the new activity fee with the same details for the Allentown which Chicago Activity fee has & linked that activity to the account. The support team was able to use the mapping document provided to aid with this. 
> 
> Late Charges:
> 
> 1. There are 2 Late Charges named as Late Charge & Late Charge 2 of the site Chicago but the Allentown has none of them. So, we have created 2 new late charges of the same name for the Allentown site & linked those with the respected newly moved accounts.
> 
> Billing Periods 
> 
> 1. There are 15 Billing Periods under the Chicago Site but we didn't find any of the Chicago account's recurring fee linked to any of the billing periods. So, we didn't create any of the Billing Period under the Allentown Site.
> 
> Taxes:
> 
> 1. There are no Taxes under the Chicago Site.
> 
> Holidays:
> 
> 1. We didn't change anything new with the Allentown Holiday Fee.
> 
> Sub Billing Cycle:
> 
> 1. There are 3 Billing Cycle under the Chicago site:
>  * Monthly
>  * NCSM 1st Cycle Accounts - 0 Account
>  * NCSM 15th Cycle Accounts  - 1 Account (N1013 NCSM- Aviem)
> 
> 2. The last 2 Billing Cycle names are also presented under the Allentown Site. So, we created only 1 new billing cycle named as Monthly of which Last Billing Cycle Date
> updated as the Last Billing Cycle Date of the Monthly Billing Cycle of the Chicago Site which is 01-June-2019. 
> We have also linked all of the Chicago accounts having the Monthly Billing cycle to the newly created Billing Cycle under the Chicago site.
> There is only 1 account of the Chicago Site having the NCSM 15th Cycle Accounts which is now linked to the Allentown's NCSM 15th Cycle Accounts.
> 
> Notes:
> 1. Please confirm everything is working correctly before sending out any invoices. Please let us know of any major changes that need to be performed by the support team.
> 2. There are no longer any accounts under the Chicago site.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2019, 13:06

@smasih @amishra @pchaudhary @tim.traylor 

After a call with Kim and Alina we found out that none of the Chicago accounts generated invoices. After discussion we found out that this was due to Chicago accounts being assigned to a new sub billing cycle (Monthly) rather than to the existing Master billing cycle. Ops recalls that the Chicago accounts did generated invoices on the Development server. So we will want the support team to confirm that what was exactly on the Development server was released to the Production Server.

 

In the end we discussed that the best solution would be the following:

 

1. Revert the Billing Cycle - Dyad Master. (I was able to complete this on the call)
1. Ops will confirm all staged fees went back into a pending status and make any modifications as needed. (Confirmed)
1. The Support team to move all of the Dyad Monthly (Sub) Billing Cycle accounts to the Dyad Master Billing Cycle.
 1. If possible can the Support team restrict sending invoices to the Allentown accounts, that already had their invoice sent to them, for just this billing cycle? If not, the invoices should be exactly the same, so we would expect little confusion.
1. The support team will need to delete the Sub Monthly Billing Cycle from Dyad, as there are now no accounts associated with it, and it would never be used.
1. Ops can then reprocess the Dyad Master Billing cycle; which now includes all of the Chicago accounts.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 28, 2019, 01:59

@kdjstudios 

> After a call with Kim and Alina we found out that **none** of the Chicago accounts generated invoices.

* I am a little confused here. As the number of accounts Merged to the Dyad site from the Chicago is `210` & the number of accounts associated with the newly created Monthly Billing Cycle is `136` only, which means that the remaining accounts should be able to create the invoices but you mentioned none of the accounts was able to.




> The Support team to move all of the Dyad Monthly (Sub) Billing Cycle accounts to the Dyad Master Billing Cycle.

* Although, we have unlinked all those `136` accounts from the newly **Monthly** sub-billing cycle & linked those to the Billing Cycle **Master** of the Dyad Site.  


> 1. If possible can the Support team restrict sending invoices to the Allentown accounts, that already had their invoice sent to them, for just this billing cycle? If not, the invoices should be exactly the same, so we would expect little confusion.


* We would like to mention here that the 2 invoices can't be the same even all of the fees & calculation are same because it differs from the **Invoice Number**. I must say that in the above case, all of the re-generated invoices will be the same except its number.

So, the thing we would like to ask to restrict the user to send the invoices are:
  * From where do you need to restrict? 
    * **Billing Cycle Page** - Then we would like to mention here that in this case, none of the invoices can be sent to anyone, not even the new generated invoices for those accounts which were linked to the *Monthly sub-billing cycle previously*. - (**1 Hour**, *We also would like to confirm that the Admin will still be able to send the invoices*)
    * **Account Billing Cycle Index Page** - This needs to be review with Tim. But still, we have an idea to put an **attribute in the invoice model**, which can distinguish between those invoices which need to be restricted. (**Take 2-3 Hours**)
    *  **Billing Cycle Index Page (Restrict only specific invoices)** - This can be the case, where you are needing the functionality to restrict to send only those invoices which will be from the accounts which were not linked to the *Monthly Sub Billing Cycle* previously because those accounts are creating the same invoice again with the different invoice number.

> The support team will need to delete the Sub Monthly Billing Cycle from Dyad, as there are now no accounts associated with it, and it would never be used.

* We have also deleted that newly created **Monthly** sub-billing cycle from the Dyad Site. 




* Please look into this & let us know if there's something else needed to complete this.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 28, 2019, 05:59

@kdjstudios Kyle, as per my discussion with the team this AM the issue was that there was a sub-billing cycle in Chicago that had a lot of accounts in it.  AFAIK, we didn't instruct Aavaz to put these accounts into the Allentown master cycle.

But, when they did the test they didn't create the sub-cycle and all of those accounts got assigned to the master.  So it looked fine in test because they had wanted those accounts moved to the master.  However, when it was run in production the sub-cycle was crated and the accounts moved there.

So we had an issue on our side in not specifying the target cycle but the team made a mistake in executing the test differently than what the production was going to be.  Also, the team did specify that the new cycle had been created for Allentown, but we didn't pick up on what that meant.

Anyway, at this point the accounts that had been moved to the monthly sub-cycle have been moved to the master and the monthly sub-cycle was removed.

Hit me up if you have questions.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 28, 2019, 08:14

@tim.traylor @pchaudhary @amishra @smasih 

Thank you for the update. So we did release something different to production then what was approved on development. Please remember team, in these scenarios you should ask instead of decide. 


I wrote:

> The support team was able to re-associate the Chicago accounts with the Master billing cycle. They however were not able to restrict the sending of invoices to only the Chicago accounts. They also mentioned that even though the same billable amounts should be on the Allentown accounts, the invoice number generated for those will be different. We would recommend if needed you still send out communication informing those clients that they will receive two invoices and to only pay the second one. Currently you should be ready to reprocess your billing. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2019, 08:30

Alina wrote:

> While doing billing I did see a discrepancy with the reports.  I did find on the Charges and Credits report it is not showing everything in the report.  For instance with the ProRate Service the report shows $-116.73 (which is Dr Bo's Diet Center) but when you look at code in the drop down it is actually $-981.09 (which is showing Dr Bo's and Commonwealth Medical).
> 
> Could you please look at and let me know if I am missing something.

I wrote:

> Thank you for notifying us. I did just run the service code report and it does show both accounts with the 6991 fee and their respective totals. I then ran the Charges and Credits report too and noticed that the summary is not pulling in both accounts, however, when I selected the 6991 code it was showing both accounts. We will have the support team look into this further and follow up. 
> 
> NOTE: Just a heads up, that this appears to be more an possible issue with the report and not with the Merge, so we may end up separating this out into a separate ticket. We will keep you posted with updates. 


@smasih @amishra @pchaudhary 

It shows both accounts under the selected code:

![image](/uploads/e44b0b1159def0db2097ab0458b5c3a0/image.png)

But it only shows the one charge under the summary(all) filter on the report. Something seems to not be working.

![image](/uploads/0b96a80bb956d0a03afb58cfe2b8ec9c/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 2, 2019, 07:54

@kdjstudios @tim.traylor 

We looked into this & below are our findings:

* The Charges & credits report is working fine.
* The discrepancy of the amount is due to the Site Merge. 
  * There was the Account Staged Fees while Merging the accounts.
  * Due to which, the fee line items of the invoice contain the same service code but of the different sites.  
  * In the above case, **COMMONWEALTH MEDICAL** account has been merged to the Dyad from Chicago & at the time of invoice creation, it contains the Account Staged Fees which was basically from the **Chicago** additional fee.
  * So, the invoice calculation & everything went well but while reporting it is causing the issue.

**Resolution**
* We can make the change from the backend & the report goes well. 
* We request you to please check the same report on the Merge2 server where we have changed the additional fee id from the backend & the report is OK now.


* Hope this helps.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 2, 2019, 11:07

@pchaudhary 

Thank you for the update. Yes please do make the change to the back end. I was able to confirm all appears to be working on the merge2 server. May we also confirm this did not happen with any other billing codes that were staged fees originally at Chicago then were transfers to the Dyad? if there are, please update those also.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 01:06

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 04:46

@kdjstudios 

We looked into this & found that there were 7 accounts listed below which do have the same effects on its last invoice, so we have fixed all of them.

* COMMONWEALTH MEDICAL
* Cambridge Square Condos
* HORWITZ LAW GROUP
* MARIA P. ZIPP, M.D.
* METROPOLITAN FIRE
* NORTH SHORE CARDIOLOGISTS
* Next Horizon Management

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2019, 09:59

@pchaudhary 

Thank you!

I wrote:

> The support team was able to find and fix the issue. May you please confirm the report is working for you now? Thank you and we look forward to your response

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 9, 2019, 11:52

closed
